### DISCUSSION TITLE
Filter metric tags at runtime

### DISCUSSION THREAD

**Discussion by gbrulotte** [open-telemetry/opentelemetry-dotnet#5682] [General]
Hi, our service is generating metrics with multiple tags and we are looking into a way to filter out some tags at runtime dynamically within the SDK. Views can only be configured at startup, unfortunately. This is offered by the OpenTelemetry Collector, but ideally, we would have a way to do this filtering within the service. I looked into the source code and I don't see any hooks available, most of the classes are internal. It is already supported for the traces within the SDK (https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/extending-the-sdk/README.md#filtering-processor), is there any reason it is not available for metrics?

Thanks for your help!

**Comment by cijothomas** [comment]
Duplicate of https://github.com/open-telemetry/opentelemetry-dotnet/issues/5583

**Comment by gbrulotte** [comment]
I will monitor the link you provided on the other issue (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricfilter), it seems to be addressing our need, thanks!
